client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -0,0 +1,132 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import FormattedHeader from 'components/formatted-header';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import Tile from 'components/tile-grid/tile';
+import TileGrid from 'components/tile-grid';
+import { addQueryArgs } from 'lib/route';
+import { getUnconnectedSiteUrl } from 'state/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { JETPACK_ONBOARDING_STEPS as STEPS } from '../constants';
+import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
+
+class JetpackOnboardingStatsStep extends React.Component {
+	componentDidUpdate() {
```

Comment:
Wouldn't `componentWillReceiveProps` be a better fit here?

Reply:
(Neither method is called on initial render BTW -- could that be a problem for the redirect from JPC?)

Reply:
> Wouldn't componentWillReceiveProps be a better fit here?

AFAIK it's recommended to keep side effects (network requests) in `componentDidUpdate` or `componentDidMount`. Isn't `componentWillReceiveProps` meant mostly for early updating state based on the new props?

Reply:
> (Neither method is called on initial render BTW -- could that be a problem for the redirect from JPC?)

An update will be called at least once because we query the onboarding settings once JPC redirects back to JPO. I did try it by hacking a custom redirect in JPC and it worked just fine. 

We can always come back and update it if necessary (it's a kind of guessing right now, because we don't have the JPC redirect yet).

Reply:
Yeah, you might be right there. It's just that `componentDidUpdate` is only called _after_ the component is re-rendered (i.e. the DOM is updated), which seems kinda late in the lifecycle here. `componentWillUpdate` maybe?

Reply:
Uh, side effects in `componentWillUpdate` aren't recommended either:

> Note that you cannot call this.setState() here; nor should you do anything else (e.g. dispatch a Redux action) that would trigger an update to a React component before componentWillUpdate() returns.

¯\\_(ツ)_/¯